[WTF] Use enum class for Safer C++ in WTF

Gumichocopengin8 opened the pull request:
#### c8eaffedffdfb28891bd1b4e4599b647b8f680e3
<pre>
[WTF] Use enum class for Safer C++ in WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=296483">https://bugs.webkit.org/show_bug.cgi?id=296483</a>

Reviewed by NOBODY (OOPS!).

This is a refactoring-only patch with no behavior changes.
This patch replaces `enum` with `enum class` for Safer C++ in WTF [1].
Some enums intentionally left unchanged due to the large scope.

[1]: <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#use-enum-classes-instead-of-old-style-enumerations">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#use-enum-classes-instead-of-old-style-enumerations</a>

* Source/JavaScriptCore/heap/StructureAlignedMemoryAllocator.cpp:
(JSC::StructureMemoryManager::StructureMemoryManager):
* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
(JSC::ExecutableAllocator::disableJIT):
(JSC::initializeJITPageReservation):
* Source/JavaScriptCore/runtime/ISO8601.cpp:
(JSC::ISO8601::checkedCastDoubleToInt128):
* Source/JavaScriptCore/runtime/ProfilerSupport.cpp:
(JSC::ProfilerSupport::ProfilerSupport):
* Source/WTF/wtf/BitVector.h:
* Source/WTF/wtf/CheckedArithmetic.h:
(WTF::operator+):
(WTF::operator-):
(WTF::operator*):
(WTF::operator/):
* Source/WTF/wtf/CheckedRef.h:
* Source/WTF/wtf/CodePtr.h:
(WTF::CodePtr::CodePtr):
(WTF::CodePtr::fromTaggedPtr):
(WTF::CodePtr::fromUntaggedPtr):
(WTF::CodePtr::retagged const):
* Source/WTF/wtf/DataLog.cpp:
(WTF::setDataFile):
* Source/WTF/wtf/FilePrintStream.cpp:
(WTF::FilePrintStream::~FilePrintStream):
* Source/WTF/wtf/FilePrintStream.h:
* Source/WTF/wtf/Function.h:
(WTF::Function&lt;Out):
* Source/WTF/wtf/FunctionPtr.h:
* Source/WTF/wtf/Gigacage.h:
(): Deleted.
* Source/WTF/wtf/HexNumber.h:
(WTF::Internal::hexDigitsForMode):
(WTF::hex):
* Source/WTF/wtf/Lock.cpp:
(WTF::Lock::unlockSlow):
(WTF::Lock::unlockFairlySlow):
* Source/WTF/wtf/LockAlgorithm.h:
(WTF::LockAlgorithm::unlock):
(WTF::LockAlgorithm::unlockFairly):
(): Deleted.
* Source/WTF/wtf/LockAlgorithmInlines.h:
(WTF::Hooks&gt;::lockSlow):
(WTF::Hooks&gt;::unlockSlow):
* Source/WTF/wtf/Locker.h:
* Source/WTF/wtf/LocklessBag.h:
* Source/WTF/wtf/MathExtras.h:
(WTF::roundUpToMultipleOfNonPowerOfTwo):
* Source/WTF/wtf/MessageQueue.h:
(WTF::MessageQueue&lt;DataType&gt;::waitForMessage):
(WTF::MessageQueue&lt;DataType&gt;::waitForMessageFilteredWithTimeout):
* Source/WTF/wtf/OSAllocator.h:
(): Deleted.
* Source/WTF/wtf/ObjectIdentifier.h:
(WTF::ObjectIdentifierGeneric::generate):
* Source/WTF/wtf/OptionSet.h:
(WTF::OptionSet::fromRaw):
* Source/WTF/wtf/PageAllocation.h:
(WTF::PageAllocation::allocate):
* Source/WTF/wtf/PageReservation.h:
(WTF::PageReservation::reserve):
(WTF::PageReservation::tryReserve):
(WTF::PageReservation::reserveWithGuardPages):
(WTF::PageReservation::tryReserveWithGuardPages):
(WTF::PageReservation::reserveAndCommitWithGuardPages):
(WTF::PageReservation::tryReserveAndCommitWithGuardPages):
* Source/WTF/wtf/PtrTag.h:
(WTF::tagNativeCodePtrImpl):
(WTF::untagNativeCodePtrImpl):
(WTF::tagCodePtrImpl):
(WTF::retagCodePtrImplHelper):
(WTF::assertIsCFunctionPtr):
(WTF::assertIsNotTagged):
(WTF::isTaggedWith):
(WTF::tagCFunctionPtrImpl):
(WTF::untagCFunctionPtrImpl):
(): Deleted.
* Source/WTF/wtf/RedBlackTree.h:
* Source/WTF/wtf/Ref.h:
(WTF::adoptRef):
* Source/WTF/wtf/RefPtr.h:
(WTF::adoptRef):
* Source/WTF/wtf/RetainPtr.h:
(WTF::adoptCF):
(WTF::adoptNS):
* Source/WTF/wtf/SentinelLinkedList.h:
(WTF::SentinelLinkedList::SentinelLinkedList):
* Source/WTF/wtf/StdLibExtras.h:
(WTF::binarySearchImpl):
(WTF::binarySearch):
(WTF::tryBinarySearch):
(WTF::approximateBinarySearch):
* Source/WTF/wtf/Threading.h:
* Source/WTF/wtf/URL.cpp:
(WTF::URL::isIPv6Address):
* Source/WTF/wtf/WTFConfig.cpp:
(WTF::Config::initialize):
* Source/WTF/wtf/WTFConfig.h:
(): Deleted.
* Source/WTF/wtf/WorkQueue.cpp:
(WTF::WorkQueue::mainSingleton):
* Source/WTF/wtf/WorkQueue.h:
* Source/WTF/wtf/dtoa/bignum-dtoa.cc:
* Source/WTF/wtf/dtoa/bignum-dtoa.h:
(): Deleted.
* Source/WTF/wtf/dtoa/double-conversion.cc:
* Source/WTF/wtf/dtoa/fast-dtoa.cc:
* Source/WTF/wtf/dtoa/fast-dtoa.h:
(): Deleted.
* Source/WTF/wtf/fast_float/ascii_number.h:
* Source/WTF/wtf/fast_float/float_common.h:
* Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp:
(WTF::OSAllocator::tryReserveAndCommit):
* Source/WTF/wtf/posix/ThreadingPOSIX.cpp:
(WTF::Thread::waitForCompletion):
* Source/WTF/wtf/text/IntegerToStringConversion.h:
(WTF::numberToStringImpl):
(WTF::numberToStringSigned):
(WTF::numberToStringUnsigned):
(WTF::writeIntegerToBufferImpl):
(WTF::writeIntegerToBuffer):
(WTF::lengthOfIntegerAsStringImpl):
(WTF::lengthOfIntegerAsString):
* Source/WTF/wtf/text/StringImpl.h:
(WTF::StringImpl::StaticStringImpl::StaticStringImpl):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::SplitResult::end const):
* Source/WTF/wtf/text/SymbolImpl.h:
(WTF::SymbolImpl::StaticSymbolImpl::StaticSymbolImpl):
* Source/WTF/wtf/unix/UnixFileDescriptor.h:
(WTF::UnixFileDescriptor::duplicate const):
* Source/WTF/wtf/win/ThreadingWin.cpp:
(WTF::Thread::waitForCompletion):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c8eaffedffdfb28891bd1b4e4599b647b8f680e3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113184 "13 style errors") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32919 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23397 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119392 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/63830 "Hash c8eaffed for PR 48523 does not build (failure)") 
| | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33571 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41482 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/119392 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/63830 "Hash c8eaffed for PR 48523 does not build (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116131 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/130/builds/33571 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/138/builds/23397 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/119392 "Hash c8eaffed for PR 48523 does not build (failure)") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/130/builds/33571 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/138/builds/23397 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63146 "Hash c8eaffed for PR 48523 does not build (failure)") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/105701 "Built successfully and passed tests") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/130/builds/33571 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/138/builds/23397 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122609 "Hash c8eaffed for PR 48523 does not build (failure)") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/111800 "Built successfully and passed tests") | [❌ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40262 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/123/builds/41482 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/122609 "Hash c8eaffed for PR 48523 does not build (failure)") | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40647 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/138/builds/23397 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/122609 "Hash c8eaffed for PR 48523 does not build (failure)") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/138/builds/23397 "Hash c8eaffed for PR 48523 does not build (failure)") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/36398 "The change is no longer eligible for processing.") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40148 "Hash c8eaffed for PR 48523 does not build (failure)") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45647 "Failed to build and analyze WebKit") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/136030 "Built successfully") | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39789 "Hash c8eaffed for PR 48523 does not build (failure)") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36503 "Found 1 new JSC stress test failure: stress/builtin-function-is-construct-type-none.js.dfg-eager (failure)") | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43122 "Hash c8eaffed for PR 48523 does not build (failure)") | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41526 "Hash c8eaffed for PR 48523 does not build (failure)") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7c8ff54009ae29c8fc4ccd508beb812caf868c49. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48523

Gumichocopengin8 committed:
[WTF] Use enum class for Safer C++ in WTF
https://bugs.webkit.org/show_bug.cgi?id=296483

Reviewed by NOBODY (OOPS!).

This is a refactoring-only patch with no behavior changes.
This patch replaces `enum` with `enum class` for Safer C++ in WTF [1].
Some enums intentionally left unchanged due to the large scope.

[1]: https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#use-enum-classes-instead-of-old-style-enumerations

* Source/JavaScriptCore/heap/StructureAlignedMemoryAllocator.cpp:
(JSC::StructureMemoryManager::StructureMemoryManager):
* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
(JSC::ExecutableAllocator::disableJIT):
(JSC::initializeJITPageReservation):
* Source/JavaScriptCore/runtime/ISO8601.cpp:
(JSC::ISO8601::checkedCastDoubleToInt128):
* Source/JavaScriptCore/runtime/ProfilerSupport.cpp:
(JSC::ProfilerSupport::ProfilerSupport):
* Source/WTF/wtf/BitVector.h:
* Source/WTF/wtf/CheckedArithmetic.h:
(WTF::operator+):
(WTF::operator-):
(WTF::operator*):
(WTF::operator/):
* Source/WTF/wtf/CheckedRef.h:
* Source/WTF/wtf/CodePtr.h:
(WTF::CodePtr::CodePtr):
(WTF::CodePtr::fromTaggedPtr):
(WTF::CodePtr::fromUntaggedPtr):
(WTF::CodePtr::retagged const):
* Source/WTF/wtf/DataLog.cpp:
(WTF::setDataFile):
* Source/WTF/wtf/FilePrintStream.cpp:
(WTF::FilePrintStream::~FilePrintStream):
* Source/WTF/wtf/FilePrintStream.h:
* Source/WTF/wtf/Function.h:
(WTF::Function<Out):
* Source/WTF/wtf/FunctionPtr.h:
* Source/WTF/wtf/Gigacage.h:
(): Deleted.
* Source/WTF/wtf/HexNumber.h:
(WTF::Internal::hexDigitsForMode):
(WTF::hex):
* Source/WTF/wtf/Lock.cpp:
(WTF::Lock::unlockSlow):
(WTF::Lock::unlockFairlySlow):
* Source/WTF/wtf/LockAlgorithm.h:
(WTF::LockAlgorithm::unlock):
(WTF::LockAlgorithm::unlockFairly):
(): Deleted.
* Source/WTF/wtf/LockAlgorithmInlines.h:
(WTF::Hooks>::lockSlow):
(WTF::Hooks>::unlockSlow):
* Source/WTF/wtf/Locker.h:
* Source/WTF/wtf/LocklessBag.h:
* Source/WTF/wtf/MathExtras.h:
(WTF::roundUpToMultipleOfNonPowerOfTwo):
* Source/WTF/wtf/MessageQueue.h:
(WTF::MessageQueue<DataType>::waitForMessage):
(WTF::MessageQueue<DataType>::waitForMessageFilteredWithTimeout):
* Source/WTF/wtf/OSAllocator.h:
(): Deleted.
* Source/WTF/wtf/ObjectIdentifier.h:
(WTF::ObjectIdentifierGeneric::generate):
* Source/WTF/wtf/OptionSet.h:
(WTF::OptionSet::fromRaw):
* Source/WTF/wtf/PageAllocation.h:
(WTF::PageAllocation::allocate):
* Source/WTF/wtf/PageReservation.h:
(WTF::PageReservation::reserve):
(WTF::PageReservation::tryReserve):
(WTF::PageReservation::reserveWithGuardPages):
(WTF::PageReservation::tryReserveWithGuardPages):
(WTF::PageReservation::reserveAndCommitWithGuardPages):
(WTF::PageReservation::tryReserveAndCommitWithGuardPages):
* Source/WTF/wtf/PtrTag.h:
(WTF::tagNativeCodePtrImpl):
(WTF::untagNativeCodePtrImpl):
(WTF::tagCodePtrImpl):
(WTF::retagCodePtrImplHelper):
(WTF::assertIsCFunctionPtr):
(WTF::assertIsNotTagged):
(WTF::isTaggedWith):
(WTF::tagCFunctionPtrImpl):
(WTF::untagCFunctionPtrImpl):
(): Deleted.
* Source/WTF/wtf/RedBlackTree.h:
* Source/WTF/wtf/Ref.h:
(WTF::adoptRef):
* Source/WTF/wtf/RefPtr.h:
(WTF::adoptRef):
* Source/WTF/wtf/RetainPtr.h:
(WTF::adoptCF):
(WTF::adoptNS):
* Source/WTF/wtf/SentinelLinkedList.h:
(WTF::SentinelLinkedList::SentinelLinkedList):
* Source/WTF/wtf/StdLibExtras.h:
(WTF::binarySearchImpl):
(WTF::binarySearch):
(WTF::tryBinarySearch):
(WTF::approximateBinarySearch):
* Source/WTF/wtf/Threading.h:
* Source/WTF/wtf/URL.cpp:
(WTF::URL::isIPv6Address):
* Source/WTF/wtf/WTFConfig.cpp:
(WTF::Config::initialize):
* Source/WTF/wtf/WTFConfig.h:
(): Deleted.
* Source/WTF/wtf/WorkQueue.cpp:
(WTF::WorkQueue::mainSingleton):
* Source/WTF/wtf/WorkQueue.h:
* Source/WTF/wtf/dtoa/bignum-dtoa.cc:
* Source/WTF/wtf/dtoa/bignum-dtoa.h:
(): Deleted.
* Source/WTF/wtf/dtoa/double-conversion.cc:
* Source/WTF/wtf/dtoa/fast-dtoa.cc:
* Source/WTF/wtf/dtoa/fast-dtoa.h:
(): Deleted.
* Source/WTF/wtf/fast_float/ascii_number.h:
* Source/WTF/wtf/fast_float/float_common.h:
* Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp:
(WTF::OSAllocator::tryReserveAndCommit):
* Source/WTF/wtf/posix/ThreadingPOSIX.cpp:
(WTF::Thread::waitForCompletion):
* Source/WTF/wtf/text/IntegerToStringConversion.h:
(WTF::numberToStringImpl):
(WTF::numberToStringSigned):
(WTF::numberToStringUnsigned):
(WTF::writeIntegerToBufferImpl):
(WTF::writeIntegerToBuffer):
(WTF::lengthOfIntegerAsStringImpl):
(WTF::lengthOfIntegerAsString):
* Source/WTF/wtf/text/StringImpl.h:
(WTF::StringImpl::StaticStringImpl::StaticStringImpl):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::SplitResult::end const):
* Source/WTF/wtf/text/SymbolImpl.h:
(WTF::SymbolImpl::StaticSymbolImpl::StaticSymbolImpl):
* Source/WTF/wtf/unix/UnixFileDescriptor.h:
(WTF::UnixFileDescriptor::duplicate const):
* Source/WTF/wtf/win/ThreadingWin.cpp:
(WTF::Thread::waitForCompletion):
